storage/innobase/os/os0file.cc

```diff
@@ -1308,6 +1308,33 @@ os_file_create_simple_func(
 	return(file);
 }
 
+/** Disable OS I/O caching on the file if the file type and server
+configuration requires it.
+@param file handle to the file
+@param name name of the file, for diagnostics
+@param mode_str operation on the file, for diagnostics
+@param type if OS_FILE_READ_WRITE_CACHED, then caching will be disabled
+unconditionally, ignored otherwise.
+@param access_type */
+static
+void
+os_file_set_nocache_if_needed(os_file_t file, const char* name,
+			      const char *mode_str, ulint type,
+			      ulint access_type)
+{
+	if (srv_read_only_mode || access_type == OS_FILE_READ_WRITE_CACHED)
+		return;
+
+	if (srv_unix_file_flush_method == SRV_UNIX_O_DIRECT
```

Comment:
Condition needs to be:

if (srv_unix_file_flush_method == SRV_UNIX_ALL_O_DIRECT
||  (type != OS_LOG_FILE 
     && (srv_unix_file_flus_method= SRV_UNIX_O_DIRECT
            || srv_unix_file_flush_method == SRV_UNIX_O_DIRECT_NO_SYNC)) {
}